charts/mlflow/templates/helpers/_deployment_commons.tpl

```diff
@@ -0,0 +1,60 @@
+{{/* mlflow deployment spec template metadata for use in all deployment manifests */}}
+{{- define "mlflow.deploymentTemplateMetadataValues" -}}
+metadata:
+  {{- with .Values.podAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    {{- include "mlflow.selectorLabels" . | nindent 4 }}
+{{- end -}}
+
+
+{{/* mlflow deployment commen template spec configuration values */}}
+{{- define "mlflow.deploymentCommonTemplateSpecValues" -}}
+{{- with .Values.image.pullSecrets }}
+imagePullSecrets:
+  {{- toYaml . | nindent 2 }}
+{{- end }}
+serviceAccountName:
+  {{- include "mlflow.serviceAccountName" . | nindent 2}}
+securityContext:
+  {{- toYaml .Values.podSecurityContext | nindent 2 }}
+{{- with .Values.nodeSelector }}
+nodeSelector:
+  {{- toYaml . | nindent 2 }}
+{{- end }}
+{{- with .Values.affinity }}
+affinity:
+  {{- toYaml . | nindent 2 }}
+{{- end }}
+{{- with .Values.tolerations }}
+tolerations:
+  {{- toYaml . | nindent 2 }}
+{{- end }}
+{{- end -}}
+
+
+{{/* mlflow tracking server container common configuration values */}}
+{{- define "mlflow.trackingServerContainerCommonValues" -}}
+securityContext:
+  {{- toYaml .Values.securityContext | nindent 2 }}
+image: "{{ .Values.image.repository }}:{{ .Chart.AppVersion }}"
```

Comment:
`.Chart.AppVersion` is `X.Y.Z` but the tags are of the form `vX.Y.Z`. And this completely ignores what's specified in the values.

```suggestion
image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default (printf "v%s" .Chart.AppVersion) }}"
```

Reply:
In my experience it is normal that the AppVersion is specified in the Chart.yaml including the `v` prefix, even if the Chart version is not. And that is what I would recommend; as the AppVersion then is exactly what is being published as tags and docker images on the (Github) repository.

For example have a look at ArgoCD:
https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/Chart.yaml